lib/pbench/agent/tool_data_sink.py

```diff
@@ -251,48 +290,56 @@ def launch(self):
                 f"{self.benchmark_run_dir}/tm/prometheus.yml:/etc/prometheus/prometheus.yml:Z",
                 "--network",
                 "host",
-                prom_reg,
+                self.prom_reg,
             ]
             try:
                 self.run = subprocess.Popen(args, stdout=prom_logs, stderr=prom_logs)
             except Exception as exc:
-                self.logger.error("Podman run process failed: '%s'", exc)
+                self.logger.error("Podman run process failed: '%s', %r", exc, args)
                 self.run = None
-                return 0
-        return 1
 
     def terminate(self):
-        if super().terminate() == 0:
-            return 0
+        """terminate - shuts down the prometheus sub-process, and creates a
+        tar ball of the data collected.
+        """
+        try:
+            super().terminate()
+        except Exception:
+            self.logger.error("Prometheus failed to terminate")
+            return
 
         self.logger.debug("Prometheus terminated")
 
         args = [
-            "tar",
+            tar_path,
             "--remove-files",
-            "--exclude",
-            "prometheus/prometheus_data.tar.gz",
-            "-zcvf",
-            f"{self.volume}/prometheus_data.tar.gz",
+            "-zcf",
+            f"{self.tool_group_dir}/prometheus_data.tar.gz",
             "-C",
             f"{self.tool_group_dir}/",
             "prometheus",
         ]
         data_store = subprocess.Popen(args)
         data_store.wait()
 
-        return 1
-
 
 class PCPCollector(BaseCollector):
     """Persistent tool data collector for tools compatible with Prometheus"""
 
     def __init__(self, *args, **kwargs):
+        """Constructor - responsible for setting up the state needed to run
+        the PCP collector.
+        """
         super().__init__(*args, **kwargs)
         self.volume = self.tool_group_dir / "pcp"
-        self.podname = None
+        benchmark_run_dir_bytes = str(self.benchmark_run_dir).encode("utf-8")
+        suffix = hashlib.md5(benchmark_run_dir_bytes).hexdigest()
+        self.podname = f"collector-{suffix}"
 
     def __test_conn(self):
```

Comment:
BROKEN: TDS waiting for pmcds to be launched and reachable before launching pmlogger, but with the changes in this PR the pmcds will never launch until after the TDS is finished its init phase for pmlogger

Reply:
Note: Comment is meant to reference __test_conn function

Reply:
Right, I ran into this and fixed it in PR #2076.  This is a good catch, and I'll backport the fix here.

Reply:
Fixed.